tests/Type/EnumTypeTest.php

```diff
@@ -64,10 +64,10 @@ public function setUp(): void
 
         $ComplexEnum = new EnumType([
```

Comment:
This test case is supposed to mirror the reference implementation. Please revert this part.

Instead, add a dedicated test method such as `testLazilyDefineValuesAsCallable()`. In addition to `getValues()` working, the test should also ensure lazyness - the callable should not be evaluated upon construction.

Reply:
Ok, can I still add a new enum type to the setUp or should it be isolated in the `testLazilyDefineValuesAsCallable`?

Reply:
Isolated.

Reply:
Done